x-pack/plugins/observability/public/plugin.ts

```diff
@@ -25,15 +32,19 @@ export class Plugin implements PluginClass<ClientSetup, ClientStart> {
       appRoute: '/app/observability',
       category: DEFAULT_APP_CATEGORIES.observability,
 
-      async mount(params: AppMountParameters<unknown>) {
+      mount: async (params: AppMountParameters<unknown>) => {
         // Load application bundle
         const { renderApp } = await import('./application');
         // Get start services
         const [coreStart] = await core.getStartServices();
 
-        return renderApp(coreStart, params);
+        return renderApp(coreStart, params, this.observabilityDataAccessService.getHandlers());
       },
     });
+
+    return {
+      chartDataFetcher: this.observabilityDataAccessService.setup(core),
```

Comment:
Would it make sense to align the naming in the external contract, e.g. to

```suggestion
      dataAccess: this.observabilityDataAccessService.setup(core),
```

?